src/main/java/org/mule/tools/devkit/sonar/checks/maven/Version.java

```diff
@@ -12,17 +12,18 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URL;
-import java.util.List;
 import java.util.function.Predicate;
-import java.util.stream.Collectors;
+import java.util.regex.Pattern;
 
 import static java.util.Optional.ofNullable;
+import static java.util.regex.Pattern.compile;
 import static org.sonar.api.internal.google.common.collect.Lists.newArrayList;
 import static org.w3c.dom.Node.ELEMENT_NODE;
 
 public class Version implements Comparable<Version> {
 
     private final String value;
+    private static final Pattern SEMANTIC_VERSION_PATTERN = compile("^(\\d(\\.\\d))(\\.\\d)?$");
```

Comment:
This would match numbers like `3.0` or `3.0.0`, but not `3.10.0` or `3.0.10`.

You can change it to something like this: `(\d(\.\d*)*)(\.\d*)?$`. (tested with http://regexr.com/)